plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php

```diff
@@ -189,11 +195,22 @@ function handleRevisionElement($node, $stageId, $fileId) {
 
 		$submissionFile->setRevision($revisionId);
 
+		if ($sourceFileAndRevision) {
+			$sourceFileId = $deployment->getFileDBId($sourceFileAndRevision[0], $sourceFileAndRevision[1]);
+			if ($sourceFileId) {
+				$submissionFile->setSourceFileId($sourceFileId);
+				$submissionFile->setSourceRevision($sourceFileAndRevision[1]);
+			}
+		}
+
 		if ($errorOccured) {
 			// if error occured, the file cannot be inserted into DB, becase
 			// genre, uploader and user group are required (e.g. at name generation).
 			$submissionFile = null;
 		} else {
+			// if the same file is already inserted, take its DB file ID
+			$DBId = $deployment->getFileDBId($fileId);
```

Comment:
It would be equivalent here if we included the `$revisionId` in this call, correct? If so, that would mean it's possible to remove the special case this PR introduces to `getFileDBId`.